contracts/ContractRegistry.sol

```diff
@@ -1,20 +1,84 @@
 pragma solidity 0.5.16;
 import "./spec_interfaces/IContractRegistry.sol";
+import "./IContractRegistryListener.sol";
 import "./WithClaimableMigrationOwnership.sol";
 import "./WithClaimableFunctionalOwnership.sol";
 
 contract ContractRegistry is IContractRegistry, WithClaimableMigrationOwnership, WithClaimableFunctionalOwnership {
 
-	mapping (string => address) contracts;
+	struct Contract {
+		address addr;
+		bool isManaged;
+	}
+	mapping (bytes32 => Contract) idToContract;
+	bytes32[] contractIds;
+
+	function setContracts(bytes32[] calldata ids, address[] calldata addrs, bool[] calldata isManaged) external onlyFunctionalOwner {
```

Comment:
modify to a single contract set (revert to the current interface keeping the id as string)